### ISSUE TITLE
Get Learning Rate SPSA & QNSPSA

### ISSUE THREAD

**Issue by Zoufalc** [Qiskit/qiskit-terra#7092]
### What is the expected enhancement?

For `SPSA` as well as `QNSPSA` you can set `learning_rate=None`.
When calling the `.settings()` property of the optimizers after running the optimization, it will return `learning_rate=None` again.
However, the effective `learning_rate` is actually [changed](https://github.com/Qiskit/qiskit-terra/blob/deebd5cce63601f5a6bb4aa762baedd9cb4876b3/qiskit/algorithms/optimizers/spsa.py#L501) by the `calibrate` function. Can `self.learning_rate` be updated accordingly?

  @Cryoris

**Comment by Cryoris** [comment]
Yes that's an interesting information, however it's not quite straightforward to resolve because
1. the learning rate is calibrated based on the function and initial point. That means it's particular to an individual minimization, but the (QN)SPSA objects can be re-used for different optimizations -- therefore we can't really store the calibrated learning rate as it'll interfere with the following runs.
2. Even if we would return the learning rate, it would be a generator (or function that returns a generator) and probably of not so much use since you can't ask it: "what functional form and what parameters do you have?"

One possible solution would be to print this information into the logger so that if the automatically calibrated learning rate is used, it would show something like
```
SPSA calibration started...
SPSA calibration finished:
- Learning rate: eta / (n + offset) ** power with eta = .., offset = .. and power = ...
- Perturbation: ... something similar ...
```
would that resolve the problem? Or are you really interested in the actual generator without the above information?

**Comment by Cryoris** [comment]
Do you want to store it to use it later in the code or just to read out the values?

**Comment by Cryoris** [comment]
Ok, then I'll add the logger for now 👍🏻

**Comment by jakelishman** [comment]
My apologies - I hadn't noticed that we were waiting for feedback.  Please comment here or open a new issue if #7109 didn't fully satisfy what you wanted.

**Comment by des137** [comment]
The changes look great. I did reach out to Julien. Thank you, @jakelishman.
